rules/template-indent.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+const stripIndent = require('strip-indent');
+const {replaceTemplateElement} = require('./fix/index.js');
+
+const MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE = 'template-indent';
+const messages = {
+	[MESSAGE_ID_IMPROPERLY_INDENTED_TEMPLATE]: 'Templates should be properly indented. Selector: {{ selector }}',
+};
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	const options = {
+		tags: ['outdent', 'dedent', 'gql', 'sql', 'html', 'styled'],
+		functions: ['dedent', 'stripIndent'],
+		selectors: [],
+		comments: ['HTML', 'indent'],
+		...context.options[0],
+	};
+
+	options.comments = options.comments.map(comment => comment.toLowerCase());
+
+	const selectors = [
+		...options.tags.map(tag => `TaggedTemplateExpression[tag.name="${tag}"] > .quasi`),
+		...options.functions.map(fn => `CallExpression[callee.name="${fn}"] > .arguments:first-child`),
+		...options.selectors,
+	];
+
+	/** @param {string} selector */
+	const getTemplateLiteralHandler = selector =>
+		/** @param {import('@babel/core').types.TemplateLiteral} node */
+		node => {
+			if (node.type !== 'TemplateLiteral') {
+				return;
+			}
+
+			const delimiter = '__PLACEHOLDER__' + Math.random();
+			const joined = node.quasis.map(q => q.value.raw).join(delimiter);
```

Comment:
Can we use `sourceCode.getText(node)`?

Reply:
I tried it, but it includes the backtick prefix and `${` suffix. Here's what I did (open to other suggestions if you know a way to avoid this problem, but also `.value.raw` seems appropriate to me:

```js
			const joined = node.quasis.map(q => sourceCode.getText(q)).join(delimiter);
```

Reply:
I don't understand, what's the problem if text including `${`, the expressions should indent too, are you considering nested template literal?

Reply:
strip-indent would get confused by the leading and trailing characters, at least with my current implementation of taking the whole template with placeholder parameters. Are you suggesting a different approach entirely?

Reply:
Switched to `sourceCode.getText(...)` which incidentally seems to respect `\r\n` newlines unlike `.raw`.